adapter_files/precice.cpp

```diff
@@ -526,14 +546,31 @@ double Precice::advance(double computedTimestepLength) {
              << "..." << endl;
       }
       // convert displacementDeltas into displacementDeltas_su2
-      for (int iVertex = 0; iVertex < vertexSize[i]; iVertex++) {
-        for (int iDim = 0; iDim < nDim; iDim++) {
-          displacementDeltas_su2[iVertex][iDim] = displacementDeltas[iVertex * nDim + iDim];
+      if (!dataTypeDisplacement) {
+        // convert displacementDeltas into displacementDeltas_su2
+        for (int iVertex = 0; iVertex < vertexSize[i]; iVertex++) {
+          for (int iDim = 0; iDim < nDim; iDim++) {
+            displacementDeltas_su2[iVertex][iDim] = displacementDeltas[iVertex * nDim + iDim];
+          }
+        }
+      } else {
+        for (int iVertex = 0; iVertex < vertexSize[i]; iVertex++) {
+          for (int iDim = 0; iDim < nDim; iDim++) {
+            displacementDeltas_su2[iVertex][iDim] =
+                displacements[iVertex * nDim + iDim] - displacements_n[iVertex * nDim + iDim];
+
+            if (solverInterface.isTimeWindowComplete()) {
+              displacements_n[iVertex * nDim + iDim] = displacements[iVertex * nDim + iDim];
+            }
+          }
```

Comment:
@uekerman I guess you are pointing to this block?
Any alternative?

Reply:
Yes. Instead of `dataTypeDisplacement` you could do a type variable for all allowed data and then a switch
```
switch(dataType):
  case: Displacement
    ...
  case: DisplacementDelta
```
This would be easier to extend in future work.
If this is too much hassle right now let's postpone this.

Reply:
Done, do you want to check it again?